src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -92,6 +91,211 @@ def _get_symbol_table(node):
+    :param int dim: the dimension index to check.
+    :param int index: the index of the range to check (0 is the \
+        lower bound, 1 is the upper bound and 2 is the step).
+    :param int value: the expected value of the literal.
+
+    :raises NotImplementedError: if the supplied argument does not \
+        have the required properties.
+
+    '''
+    _check_args(array, dim)
+
+    if not isinstance(index, int):
+        raise TypeError(
+            "method _check_array_range_literal 'index' argument should be an "
+            "int type but found '{0}'.".format(type(index).__name__))
+
+    if index < 0 or index > 2:
+        raise ValueError(
+            "method _check_array_range_literal 'index' argument should be "
+            "0, 1 or 2 but found '{0}'.".format(index))
+
+    if not isinstance(value, int):
+        raise TypeError(
+            "method _check_array_range_literal 'value' argument should be an "
+            "int type but found '{0}'.".format(type(value).__name__))
+
+    # The first child of the array (index 0) relates to the first
+    # dimension (dim 1), so we need to reduce dim by 1.
+    literal = array.children[dim-1].children[index]
+
+    if (isinstance(literal, Literal) and
+            literal.datatype == DataType.INTEGER and
+            literal.value == str(value)):
+        return True
+    return False
+
+
+def _is_range_full_extent(my_range):
+    '''Utility function to check whether a Range object is equivalent to a
+    ":" in Fortran array notation. The PSyIR representation of "a(:)"
+    is "a(lbound(a,1):ubound(a,1):1). Therefore, for array a index 1,
+    the lower bound is compared with "lbound(a,1)", the upper bound is
+    compared with "ubound(a,1)" and the step is compared with 1.
+
+    If everything is OK then this routine silently returns, otherwise
+    an exception is raised by one of the functions
+    (_check_bound_is_full_extent or _check_array_range_literal) called by this
+    function.
+
+    This routine is only in fparser2.py until #717 is complete as it
+    is used to check that array syntax in a where statement is for the
+    full extent of the dimension. Once #717 is complete this routine
+    can be removed.
+
+    :param my_range: the Range node to check.
+    :type my_range: :py:class:`psyclone.psyir.node.Range`
+
+    '''
+
+    array = my_range.parent
+    # The array index of this range is determined by its position in
+    # the array list (+1 as the index starts from 0 but Fortran
+    # dimensions start from 1).
+    dim = array.children.index(my_range) + 1
+    # Check lower bound
+    is_lower = _is_bound_full_extent(
+        array, dim, BinaryOperation.Operator.LBOUND)
+    # Check upper bound
+    is_upper = _is_bound_full_extent(
+        array, dim, BinaryOperation.Operator.UBOUND)
+    # Check step (index 2 is the step index for the range function)
+    is_step = _is_array_range_literal(array, dim, 2, 1)
+    return (is_lower and is_upper and is_step)
```

Comment:
pylint complains that the parentheses are superfluous!

Reply:
Oops, missed that one.